### PR TITLE
Add environment variable for using Integrated Vectorization

### PR DESCRIPTION
**Pull request by komalg1**
## Purpose
Required by [#321](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321)

* This PR adds an environment variable 'AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION' for choosing between Pull vs Push approach to index data to Search.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* The default value for AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION is set to 'false'

### CODE REVIEW

File: code/backend/pages/01_Ingest_Data.py [hunk 1]
```diff
@@ -122,6 +123,7 @@ def upload_file(bytes_data: bytes, file_name: str, content_type: Optional[str] =
             overwrite=True,
             content_settings=ContentSettings(content_type=content_type + charset),
         )
+        blob_client.set_blob_metadata({"source_url": file_name})
```
Comment (komalg1) [inline code/backend/pages/01_Ingest_Data.py:126]: added this for url embedding using pull not directly related to adding the env variable

File: code/backend/pages/01_Ingest_Data.py [hunk 2]
```diff
@@ -155,6 +157,7 @@ def upload_file(bytes_data: bytes, file_name: str, content_type: Optional[str] =
...
```
Comment (komalg1) [inline code/backend/pages/01_Ingest_Data.py:160]: same for this as above

File: code/backend/batch/utilities/helpers/EnvHelper.py [hunk 3]
```diff
@@ -158,6 +158,10 @@ def __init__(self, **kwargs) -> None:
         self.LOAD_CONFIG_FROM_BLOB_STORAGE = self.get_env_var_bool(
             "LOAD_CONFIG_FROM_BLOB_STORAGE"
         )
+        # Integrated Vectorization
+        self.USE_INTEGRATED_VECTORIZATION = (
```
Comment (adamdougal) [inline code/backend/batch/utilities/helpers/EnvHelper.py:162]: I'm wondering if it's worth grouping this with the `AZURE_SEARCH` env vars? E.g. `AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION`. WDYT?

File: code/backend/pages/01_Ingest_Data.py (hunk 1 above)
Comment (adamdougal) [reply code/backend/pages/01_Ingest_Data.py:126]: I'm not sure I understand what thius change is for.

Also, I think you can add the metadata in the `blob_client.upload_blob()` function call.

File: code/backend/pages/01_Ingest_Data.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
     UserDelegationKey,
 )
 import urllib.parse
+import urllib.request
```
Comment (adamdougal) [inline code/backend/pages/01_Ingest_Data.py:17]: Is this needed? I can't see any additions using it.

File: infra/main.bicep [hunk 5]
```diff
@@ -177,6 +177,8 @@ param authType string = 'keys'
 ])
 param hostingModel string = 'container'
 
+@description('Use Integrated Vectorization')
+param use_int_vectorization bool = false
```
Comment (adamdougal) [inline infra/main.bicep:181]: Same as `EnvHelper`, should we group this with Azure Search params?

File: infra/main.bicep [hunk 6]
```diff
@@ -858,4 +860,5 @@ output USE_KEY_VAULT bool = useKeyVault
 output AZURE_APP_SERVICE_HOSTING_MODEL string = hostingModel
 output FRONTEND_WEBSITE_NAME string = hostingModel == 'code' ? web.outputs.FRONTEND_API_URI : web_docker.outputs.FRONTEND_API_URI
 output ADMIN_WEBSITE_NAME string = hostingModel == 'code' ? adminweb.outputs.WEBSITE_ADMIN_URI : adminweb_docker.outputs.WEBSITE_ADMIN_URI
+output USE_INTEGRATED_VECTORIZATION bool = use_int_vectorization
```
Comment (ross-p-smith) [inline infra/main.bicep:863]: If we are following the pattern I think we need AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION

File: infra/main.bicep (hunk 5 above)
Comment (cecheta) [inline infra/main.bicep:181]: Nit: Should be camel case, and have a new line underneath

File: infra/main.bicep [hunk 7]
```diff
@@ -177,6 +177,8 @@ param authType string = 'keys'
 ])
 param hostingModel string = 'container'
 
+@description('Use Integrated Vectorization')
```
Comment (ross-p-smith) [inline infra/main.bicep:180]: Can we keep this with the other AI Search params?

File: code/backend/pages/01_Ingest_Data.py (hunk 4 above)
Comment (cecheta) [reply code/backend/pages/01_Ingest_Data.py:17]: Surprised a precommit didn't catch this

File: code/backend/batch/utilities/helpers/EnvHelper.py (hunk 3 above)
Comment (komalg1) [reply code/backend/batch/utilities/helpers/EnvHelper.py:162]: yes it should be. Updated

File: infra/main.bicep (hunk 7 above)
Comment (komalg1) [reply infra/main.bicep:180]: Updated

File: code/backend/batch/utilities/helpers/EnvHelper.py [hunk 8]
```diff
@@ -52,6 +52,11 @@ def __init__(self, **kwargs) -> None:
         self.AZURE_SEARCH_CONVERSATIONS_LOG_INDEX = os.getenv(
             "AZURE_SEARCH_CONVERSATIONS_LOG_INDEX", "conversations"
         )
+        # Integrated Vectorization
+        self.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION = (
```
Comment (cecheta) [inline code/backend/batch/utilities/helpers/EnvHelper.py:56]: There's actually a function for this below, `get_env_var_bool`

File: infra/main.bicep [hunk 9]
```diff
@@ -69,6 +69,9 @@ param azureSearchTitleColumn string = 'title'
 @description('Url column')
 param azureSearchUrlColumn string = 'url'
 
+@description('Use Azure Search Integrated Vectorization')
```
Comment (cecheta) [inline infra/main.bicep:72]:
```suggestion
@description('Use Azure Search Integrated Vectorization (Not yet implemented')
```

We should probably have some kind of note that this isn't available yet, to not cause confusion

File: code/backend/batch/utilities/helpers/EnvHelper.py (hunk 8 above)
Comment (komalg1) [reply code/backend/batch/utilities/helpers/EnvHelper.py]: Updated

File: infra/main.bicep (hunk 9 above)
Comment (komalg1) [reply infra/main.bicep:72]: Great point

File: infra/main.bicep [hunk 10]
```diff
@@ -69,6 +69,12 @@ param azureSearchTitleColumn string = 'title'
 @description('Url column')
 param azureSearchUrlColumn string = 'url'
 
+@description('Use Azure Search Integrated Vectorization (Not yet implemented)')
+@allowed([
```
Comment (komalg1) [inline infra/main.bicep:73]: Also added false as the only allowed value till work is in progress

**Review by cecheta — APPROVED** [review]
Nice :)

**Review by cecheta — APPROVED** [review]

**Pull request merged**